src/main/java/com/google/sps/servlets/CalendarServlet.java

```diff
@@ -13,28 +24,77 @@
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    SoyFileSet sfs = SoyFileSet
-        .builder()
-        .add(new File("../../src/main/java/templates/calendar.soy"))
-        .build();
-    SoyTofu tofu = sfs.compileToTofu();
-
-    String out = tofu.newRenderer("templates.calendar.calendarPage").render();
-    response.getWriter().println(out);
+    String groupId = getParameter(request, GROUP_ID_PROPERTY);
+
+    if (Strings.isNullOrEmpty(groupId)){
+      ServletUtils.printBadRequestError(response, CALENDAR_BAD_REQUEST_MESSAGE);
+      return;
+    }
+
+    String calendarId = "";
+    try {
+      calendarId = ServletUtils.getGroupProperty(groupId, GROUP_CALENDARID_PROPERTY);
+    } catch (Exception entityError) {
+      ServletUtils.printBadRequestError(response, ENTITY_ERROR_MESSAGE);
+      return;
+    }
+
+    String out = SoyRendererUtils.getOutputString(CALENDAR_SOY_FILE, CALENDAR_TEMPLATE_NAMESPACE,
+        ImmutableMap.of("calendarId", calendarId, "timezone", TIMEZONE));
+
+      response.setContentType(CONTENT_TYPE_HTML);
+      response.getWriter().println(out);
   }
 
   /**
-   * Create the Calendar page for the group being created
+   * Create a Calendar for the group being created. Query string must contain groupId.
    */
+  @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    //TODO: Implement once groups are implemented
-    throw new UnsupportedOperationException();
+    String groupId = getParameter(request, GROUP_ID_PROPERTY);
+
+    if (Strings.isNullOrEmpty(groupId)){
+      ServletUtils.printBadRequestError(response, CALENDAR_BAD_REQUEST_MESSAGE);
+      return;
+    }
+
+    try {
+      String calendarId = createCalendar(groupId);
+      setGroupCalendarId(groupId, calendarId);
+
+      response.setContentType(CONTENT_TYPE_PLAIN);
+      response.getWriter().println(calendarId);
+    } catch (Exception entityError) {
+      response.getWriter().println(ENTITY_ERROR_MESSAGE);
+    }
+  }
+
+  /**
+   * Return the Calendar ID of the created calendar.
+   */
+  @VisibleForTesting
+  public String createCalendar(String groupId) throws IOException, EntityNotFoundException {
+    Calendar calendar = new Calendar()
+        .setSummary(ServletUtils.getGroupProperty(groupId, GROUP_NAME_PROPERTY)).setTimeZone(TIMEZONE);
+
+    com.google.api.services.calendar.Calendar service = getCalendarService();
+
+    Calendar createdCalendar = service.calendars().insert(calendar).execute();
+
+    return createdCalendar.getId();
+  }
+
+  private void setGroupCalendarId(String groupId, String calendarId) throws EntityNotFoundException {
+    Entity groupEntity = ServletUtils.getGroupEntity(groupId);
+    groupEntity.setProperty(GROUP_CALENDARID_PROPERTY, calendarId);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
you can simplify this into one line:

`DatastoreServiceFactory.getDatastoreService().put(groupEntity);`